### DISCUSSION TITLE
Question on how to change the link value in DetailList Control

### DISCUSSION THREAD

**Discussion by AstroNorman** [microsoft/powercat-creator-kit#57] [Q&A] [answered]
How do I replace all the link value populating the table with Text("🔗")
<img width="179" alt="table data" src="https://user-images.githubusercontent.com/106515086/177477982-fc75fb1d-ec4e-4f8f-b4ed-05b2416774c0.png">

**Comment by Ramakrishnan24689** [comment, marked as answer]
Hi @AstroNorman,

Thanks for reaching out.

Details list supports displaying [Fluent UI Icons](https://developer.microsoft.com/en-us/fluentui#/styles/web/icons).

![image](https://user-images.githubusercontent.com/11135056/177540699-bdd190e3-4ed3-4519-af2b-9aebe025efd0.png)

**More on this [here ](https://github.com/microsoft/powercat-code-components/tree/main/DetailsList#image--clickable-image)**

Inorder to replace links with Icons. Follow below steps.

1) Customize the input collection to specify the icon to be provided as a additional column.  
Incase you want these icons to be clickable, add a RecordKey, so that, during on click events, the specific record can be uniquely identified using Lookup Formula (More on this - Step 3).

**Items** property, for example, will look like this, where LinksIcon is a **column name** and **value** is icon:_FluentUIIconName_.
_Example_
`AddColumns(
    '<<Collection/DataSource>>',
    "LinksIcon",
    "icon:Link",
    /*Optional, Icon color*/
    "IconColor",
    "rgb(0, 183, 195)",
    "RecordKey",
    /*Unique ID*/
    ID
)`

2) **columns_items** property, will look like below incase you want icons to be clickable. 
_Example_
`Table(
    {
        ColName: "LinksIcon",
        ColDisplayName: "Links",
        ColCellType: "clickableimage",
        ColIsBold: true,
        ColResizable: true,
        ColTagColorColumn: "IconColor"
    }
    /*More columns will follow*/
)`


3) Considering the icons here are link icons, you may want to re-direct the users on click over icon.

**OnChange** property, for example will look like
_Example_

`If(
    Self.EventName = "CellAction",
    /*Redirection or any other operation*/
    With(
        {
            URL: LookUp(
                '<<Collection/DataSource here>>',
                ID = Self.EventRowKey
            ).Links
        },
        Launch(URL)
    )
)`
